tests/Doctrine/Tests/Common/Proxy/ProxyMagicMethodsTest.php

```diff
@@ -285,6 +285,24 @@ public function testInheritedMagicSleep()
         self::assertSame('defaultValue', $unserialized->nonSerializedField, 'Field was not returned by "__sleep"');
     }
 
+    public function testInheritedMagicSleepTypehinted()
+    {
+        $proxyClassName = $this->generateProxyClass(MagicSleepClassTypehinted::class);
+        $proxy          = new $proxyClassName();
+
+        $sleepMethod = '__sleep';
+
+        $proxyReflection = new ReflectionClass($proxy);
+        self::assertTrue($proxyReflection->hasMethod($sleepMethod), $sleepMethod . 'method doesn\'t exist');
+
+        $methodReflection = $proxyReflection->getMethod($sleepMethod);
+        self::assertSame(
+            $methodReflection->getReturnType()->getName(),
+            'array',
+            $sleepMethod . ' method has lost its return type hint'
+        );
```

Comment:
Is deprecated method better than nothing? :wink: 
```suggestion
        $returnTypeReflection = $methodReflection->getReturnType();
        self::assertInstanceOf(\ReflectionType::class, $returnTypeReflection, 'Got unexpected return type reflection');
        if ($returnTypeReflection instanceof \ReflectionNamedType) {
            self::assertSame(
                $methodReflection->getReturnType()->getName(),
                'array',
                $sleepMethod . ' method has lost return type hint'
            );
        } else {
            self::assertSame(
                (string) $returnTypeReflection,
                'array',
                $sleepMethod . ' method has lost return type hint'
            );
        }
```